src/Kerbero.Infrastructure/NukiAuthentication/Repositories/NukiAccountPersistentRepository.cs

```diff
@@ -51,12 +51,12 @@ public async Task<Result<NukiAccount>> Create(NukiAccount nukiAccount)
 		}
 	}
 	
-	public Result<NukiAccount> GetAccount(int kerberoAccountId)
+	public async Task<Result<NukiAccount>> GetAccount(int kerberoAccountId)
 	{
 		try
 		{
-			var res = _dbContext.NukiAccounts.First(c => c.Id == kerberoAccountId);
-			return Result.Ok(res);
+			var res = await _dbContext.NukiAccounts.FindAsync(kerberoAccountId);
+			return res is null ? Result.Fail(new UnauthorizedAccessError()) : Result.Ok(res);
```

Comment:
```suggestion
			return res is null ? Result.Fail(new UnauthorizedAccessError()) : Result.Ok(res);
```
why UnauthorizedAccess if you do not find a result, it seems like a "not found error"

Reply:
Because you're searching for an Account